AWS Dev Day blogpost (#568)
ordina-jworks/ordina-jworks.github.io: _posts/2019-11-13-AWS-Dev-Day-2019.md

---
layout: post
authors: [yolan_vloeberghs]
title: 'AWS Dev Day 2019'
image: /img/aws-dev-day-2019-banner.png
tags: [AWS, Cloud, Conference]
category: Conference
comments: true
---
# AWSome Dev Day 2019!

>The AWS Dev Day is a free conference day hosted by Amazon where architects and developers from Amazon and other companies talk about the most popular AWS services, the newest features and everything related to cloud development. These sessions explain a lot about what AWS has to offer and you will see live demos on how you can implement them in your own use cases. 

> This year, AWS decided to host their annual Dev Day conference in Belgium, more specifically Mechelen. This was very convenient us, as our headquarters are located in Mechelen aswell. This means that there was a very good reason for Ordina consultants to attend the event. We went with a small group of 8 people to make sure we didn't miss the latest AWS news. We were treated with delicious food & drinks throughout the day to make sure their guests were happy.

Reviewer: AWS hosts devdays across the world every year. This was just the BE event.
So a more correct wording would be:
Amazon held their Belgian devdays in Mechelen this year. This was very convenient for us as the Ordina Belgium HQ is also located in Mechelen.

> This year, AWS decided to host their annual Dev Day conference in Belgium, more specifically Mechelen. This was very convenient us, as our headquarters are located in Mechelen aswell. This means that there was a very good reason for Ordina consultants to attend the event. We went with a small group of 8 people to make sure we didn't miss the latest AWS news. We were treated with delicious food & drinks throughout the day to make sure their guests were happy.

Reviewer: Amazon hosts devdays all over the world each year so change this intro to not make it sound like this was the only one.

> This year, AWS decided to host their annual Dev Day conference in Belgium, more specifically Mechelen. This was very convenient us, as our headquarters are located in Mechelen aswell. This means that there was a very good reason for Ordina consultants to attend the event. We went with a small group of 8 people to make sure we didn't miss the latest AWS news. We were treated with delicious food & drinks throughout the day to make sure their guests were happy.

Reviewer: ```suggestion
This year the Belgian AWS dev day was hosted in Mechelen. This was very convenient to us as our headquarters are located in Mechelen as well. Since a big focus of Jworks is on AWS there was a very good reason for us to attend the event. 8 people attended to make sure we  stay up to date with the latest AWS trends and technologies. We were treated with delicious food & drinks throughout the day.
```

# Table of contents

# CI/CD for Modern Applications by Sebastian Stormacq

### IAC
In an application development environment, it is important to maintain your Infrastructure as Code (IaC) to allow for repeatibility, and doing so in a declarative way. Doing it in an imperative way does not allow for deltas and requires your system to be rebuilt from scratch every time your pipeline gets triggered, which results in worse impact for your end users.

> There are a number of ways to provide your infrastructure as a code declaratively. We have Terraform, Cloudformation, etc. But Amazon has a new kid on the block since this year: the [Cloud Development Kit (CDK)](https://aws.amazon.com/cdk/){:target="_blank" rel="noopener noreferrer"}. This allows developers to provide your infrastructure in your favorite programming language. You can communicate with AWS in a different way other than the AWS CLI or the Web Console, as you can now create services as you need them and when you need them.

Reviewer: ```suggestion
There are a number of ways to provide your infrastructure as a code declaratively. We have Terraform, Cloudformation, etc. But Amazon has a new kid on the block since this year: the [Cloud Development Kit (CDK)](https://aws.amazon.com/cdk/){:target="_blank" rel="noopener noreferrer"}. This allows developers to provide your infrastructure in their favorite programming language. You can communicate with AWS in a way other than the AWS CLI or the Web Console, as you can now create services as you need them and when you need them.
```

> All of this gives you the best toolkit to NOT create snowflake servers.

Reviewer: Maybe explain what a snowflake server is

> All of this gives you the best toolkit to NOT create snowflake servers.

Reviewer: A word of explanation what a snowflake server is

The CDK also has a CLI which can be installed by running `npm install -g aws-cdk`. 

### Pipeline
Amazon's pipeline service is called [AWS CodePipeline](https://aws.amazon.com/codepipeline/){:target="_blank" rel="noopener noreferrer"}. This service allows you to create a pipeline which executes all the necessary steps to automate your releases and to provide infrastructure updates. This allows you to quickly provide new releases and deliver new features to your end users. You can integrate this service with other external services (ex. Github, where your repository is hosted). 

> To effectively build your application, you can make use of [AWS CodeBuild](https://aws.amazon.com/codebuild/){:target="_blank" rel="noopener noreferrer"}. This is their continuous integration service that packages your application by building it and running the tests. Pricing is very effective, as you will only pay by the minute that you trigger a build. CodeBuild automatically scales and can process multiple builds at the same time, so you won't have to deal with build queues. 

Reviewer: This is the AWS continious integration service

> To effectively build your application, you can make use of [AWS CodeBuild](https://aws.amazon.com/codebuild/){:target="_blank" rel="noopener noreferrer"}. This is their continuous integration service that packages your application by building it and running the tests. Pricing is very effective, as you will only pay by the minute that you trigger a build. CodeBuild automatically scales and can process multiple builds at the same time, so you won't have to deal with build queues. 

Reviewer: as you will only pay by the minute when you trigger a build.

### Deploy
[AWS CodeDeploy](https://aws.amazon.com/codedeploy/){:target="_blank" rel="noopener noreferrer"} is a deployment service which basically allows you to deploy (almost) anywhere, including (but not limited to) on-premise servers. This service makes it easier to provide updates to your end users and to avoid downtime during the rollout of a new update. 

> One of the most impressive features is the Blue/Green deployment. This splits the traffic going to your application into traffic to your ortiginal environment and an other environment. The best use case for this would be during a new production release. You have a new release that you want to deploy? Simply let 10% of your traffic go to that release to make sure everything works right. After monitoring the release, you can choose to either revert or to fully allow traffic to go to the newest release.

Reviewer: to your original environment and the new environment

### Conclusion

> Amazon provides a lot of tools that help a project to manage their builds and deploys, making sure that everything works according to plan. We were definitely a fan of their new Cloud Development Kit and their Blue/Green deployment is able to stabilize and fully control the rollout of a new deployment.

Reviewer: Amazon provides a lot of tools that help you manage your builds and deploys

> Amazon provides a lot of tools that help a project to manage their builds and deploys, making sure that everything works according to plan. We were definitely a fan of their new Cloud Development Kit and their Blue/Green deployment is able to stabilize and fully control the rollout of a new deployment.

Reviewer: We are definately fans

> Amazon provides a lot of tools that help a project to manage their builds and deploys, making sure that everything works according to plan. We were definitely a fan of their new Cloud Development Kit and their Blue/Green deployment is able to stabilize and fully control the rollout of a new deployment.

Reviewer: and their Blue/Green deployment option

> Amazon provides a lot of tools that help a project to manage their builds and deploys, making sure that everything works according to plan. We were definitely a fan of their new Cloud Development Kit and their Blue/Green deployment is able to stabilize and fully control the rollout of a new deployment.

Reviewer: ```suggestion
Amazon provides a lot of tools that help a project to manage their builds and deploys, making sure that everything works according to plan. We are fans of their new Cloud Development Kit and their Blue/Green deployment functionality which is able to stabilize and fully control the roll out of a new deployment.
```

# Integrate your front end apps with serverless backend in the cloud by Sebastian Stormacq

> Serverless is a big trend nowadays and rightfully so, as it does a lot of stuff for you. The biggest pros of serverless development are that you no longer need to provision or manage your own server, it automatically scales with the amount of usage, it has built in fault tolerance and fallback methods and last but not least, you only pay for the value. This means that if you are not using your lambda, you are not paying for it!

Reviewer: Serverless is  a big trend nowadays and rightfully so as it takes away a lot of operational work from you.

> Serverless is a big trend nowadays and rightfully so, as it does a lot of stuff for you. The biggest pros of serverless development are that you no longer need to provision or manage your own server, it automatically scales with the amount of usage, it has built in fault tolerance and fallback methods and last but not least, you only pay for the value. This means that if you are not using your lambda, you are not paying for it!

Reviewer: you only pay for the actual usage

> Serverless is a big trend nowadays and rightfully so, as it does a lot of stuff for you. The biggest pros of serverless development are that you no longer need to provision or manage your own server, it automatically scales with the amount of usage, it has built in fault tolerance and fallback methods and last but not least, you only pay for the value. This means that if you are not using your lambda, you are not paying for it!

Reviewer: ```suggestion
Serverless is a big trend nowadays and rightfully so! The biggest pros of serverless are that you no longer need to provision or manage your own servers, it automatically scales with usage, it has built in fault tolerance and fallback methods and last but not least, you only pay what you actually use in compute resources. This means that if you are not using your lambda, you are not paying for it!
```

> Sebastian filled the session mostly with a demo where he prepared a React app to communicate with his lambda's. 

Reviewer: ```suggestion
Sebastian filled the session mostly with a demo in which he built a React app to communicate with his lambda's. 
```

> To provide his infrastructure (auth, GraphQL API etc.), he used the [AWS Amplify CLI](https://github.com/aws-amplify/amplify-cli){:target="_blank" rel="noopener noreferrer"} which is a new CLI tool which can add cloud features to your web or mobile application. For example, by simply typing `amplify add auth` and `amplify push`, you have created your own user authentication provider (Cognito). And if you want a GraphQL API (AWS AppSync) set up, you just have to type `amplify add api` followed by `amplify push`. This creates an API and stores its data on a DynamoDB database.

Reviewer: it's

> To provide his infrastructure (auth, GraphQL API etc.), he used the [AWS Amplify CLI](https://github.com/aws-amplify/amplify-cli){:target="_blank" rel="noopener noreferrer"} which is a new CLI tool which can add cloud features to your web or mobile application. For example, by simply typing `amplify add auth` and `amplify push`, you have created your own user authentication provider (Cognito). And if you want a GraphQL API (AWS AppSync) set up, you just have to type `amplify add api` followed by `amplify push`. This creates an API and stores its data on a DynamoDB database.

Reviewer: ```suggestion
To provide his infrastructure (auth, GraphQL API etc.), he used the [AWS Amplify CLI](https://github.com/aws-amplify/amplify-cli){:target="_blank" rel="noopener noreferrer"} which is a new CLI tool that can add cloud features to your web or mobile application. For example, by simply typing `amplify add auth` and `amplify push`, you have created your own user authentication provider (Cognito). If you want to set up a GraphQL API (AWS AppSync), you just have to type `amplify add api` followed by `amplify push`. This creates an API and stores its data on a DynamoDB database.
```

> To finally deploy his application, he simply used an S3 bucket that acts as a web server. After all, React just builds static HTML / CSS / JS files which need a web server (Nginx, Apache) in order to be displayed. Using an S3 is very cost effective and doesn't require anything else other than a bucket. So no linux configuration, load balancer setup & costs, ... . By using `amplify add hosting` and `amplify publish`, these steps are automatically done and doesn't require us to do anything else. You can also enable production mode, which enables AWS CloudFront integration to send your website to edge locations, so you can provide faster loading times to your users from all over the world. 

Reviewer: Using an S3 bucket or Using S3

> To finally deploy his application, he simply used an S3 bucket that acts as a web server. After all, React just builds static HTML / CSS / JS files which need a web server (Nginx, Apache) in order to be displayed. Using an S3 is very cost effective and doesn't require anything else other than a bucket. So no linux configuration, load balancer setup & costs, ... . By using `amplify add hosting` and `amplify publish`, these steps are automatically done and doesn't require us to do anything else. You can also enable production mode, which enables AWS CloudFront integration to send your website to edge locations, so you can provide faster loading times to your users from all over the world. 

Reviewer: ```suggestion
To finally deploy his application, he simply used an S3 bucket that acts as a web server. After all, React just builds static HTML / CSS / JS files which need a web server (Nginx, Apache) in order to be displayed. Using S3 is very cost effective and doesn't require anything else other than a bucket and correct bucket configuration. So no linux configuration, load balancer setup & costs, ... . By using `amplify add hosting` and `amplify publish`, these steps are automatically done and it doesn't require us to do anything else. You can also enable production mode, which enables AWS CloudFront integration to send your website to edge locations, so you can provide faster loading times to your users from all over the world. 
```

## Conclusion
Amplify seems like a really handy CLI tool if you enjoy working with AWS. The commands are very similar to each other, and really simplifies deploying your application or integrating your application with AWS components. A new feature that has been introduced to Amplify is Predictions, which provides a solution for Artificial Intelligence & Machine Learning cloud services to let your application make use of. Overall, the future for Amplify is very bright and I can see it growing more as more developers adopt it in their current workflow. After all, who doesn't like to just type one command and everything is taken care of?

# Breaking the Monolith: road to containerizing your app

> A monolith is basically one big application that handles everything that your project needs to do. From security to having all your services in one package. This is simple, has no over-engineering (on the architecture part), faster communication as everything happens in-memory and you have one code base. BUT, monoliths grow old as well, and the older it gets, the more code it has. Your developers are also required to learn more business logic and your deployments are all or nothing, meaning that, if your application goes down, your whole infrastructure is down, which isn't interesting for an application where a lot of people depend on it.

Reviewer: you can add that monoliths scale pretty poorly and are tightly coupled systems

> Dividing your monolith into a microservice architecture improves your workflow by splitting up your team to work on each microservice. This also helps your architecture in a way that all those microservices each have their own single responsibility. However, splitting a monolith is not a simple task and requires a lot of work to make sure you split it up the right way.

Reviewer: ```suggestion
Dividing your monolith into a microservice architecture improves your workflow by splitting up your team to work on each microservice. This also helps your architecture in a way that each microservice has their own single responsibility. However, splitting a monolith is not a simple task and requires a lot of work to make sure you split it up the right way.
```

Once you have your microservices, the goal is to containerize each service.

## Fargate
[Fargate](https://aws.amazon.com/fargate/){:target="_blank" rel="noopener noreferrer"} is a service that helps you with this. It's basically a serverless implementation for containers, meaning that it manages servers and clusters for you. Simply provision your container and Fargate does the rest for you by using Amazon ECS to provision your container on the cloud. It's a really useful tool if you don't want to manage your own server or cluster. 

## Codebase, config, dependencies, ...

> The rest of the session was a very interesting take on how you actually split up your microservice. It's useful to have multiple codebases per service with their own deploy pipeline to deploy to your environments. 

Reviewer: he rest of the session was a very interesting take on how you actually split up your monolith into microservices.

> The rest of the session was a very interesting take on how you actually split up your microservice. It's useful to have multiple codebases per service with their own deploy pipeline to deploy to your environments. 

Reviewer: ```suggestion
The rest of the session was a very interesting take on how you actually can split up your monolith into microservices. It's useful to have multiple codebases per service with their own deploy pipeline to deploy to your environments. 
```

> It's also a good practice to isolate your application from your dependencies. You also should explicitly say which dependencies you want and which version that you want. This helps you to build and create immutable images with explicitly defined dependencies, so you can run the same image in multiple environments.

Reviewer: You should also

---
layout: post
authors: [yolan_vloeberghs]
title: 'AWS Dev Day 2019'
image: /img/aws-dev-day-2019-banner.png
tags: [AWS, Cloud, Conference]
category: Conference
comments: true
---
# AWSome Dev Day 2019!

>The AWS Dev Day is a free conference day hosted by Amazon where architects and developers from Amazon and other companies talk about the most popular AWS services, the newest features and everything related to cloud development. These sessions explain a lot about what AWS has to offer and you will see live demos on how you can implement them in your own use cases. 

> Amazon held their Belgian devdays in Mechelen this year. This was very convenient for us as the Ordina Belgium HQ is also located in Mechelen. This means that there was a very good reason for Ordina consultants to attend the event. We went with a small group of 8 people to make sure we didn't miss the latest AWS news. We were treated with delicious food & drinks throughout the day to make sure their guests were happy.

Author: Ik denk dat dit Dev Day moet zijn? :D

> Amazon held their Belgian devdays in Mechelen this year. This was very convenient for us as the Ordina Belgium HQ is also located in Mechelen. This means that there was a very good reason for Ordina consultants to attend the event. We went with a small group of 8 people to make sure we didn't miss the latest AWS news. We were treated with delicious food & drinks throughout the day to make sure their guests were happy.

Reviewer 2: Correct. :)

### Conclusion
Amplify seems like a really handy CLI tool if you enjoy working with AWS. The commands are very similar to each other, and really simplifies deploying your application or integrating your application with AWS components. A new feature that has been introduced to Amplify is Predictions, which provides a solution for Artificial Intelligence & Machine Learning cloud services to let your application make use of. Overall, the future for Amplify is very bright and I can see it growing more as more developers adopt it in their current workflow. After all, who doesn't like to just type one command and everything is taken care of?

> # Breaking the Monolith: road to containerizing your app

Reviewer 2: Speaker is missing from the title?

Author: Whoops, will fix
